src/wp-includes/class-wp-scripts.php

```diff
@@ -133,6 +133,24 @@ class WP_Scripts extends WP_Dependencies {
 	 */
 	private $type_attr = '';
 
+	/**
+	 * Holds a mapping of dependents (as handles) for a given script handle.
+	 * Used to optimize recursive dependency tree checks.
+	 *
+	 * @since 6.3.0
+	 * @var array
+	 */
+	private $dependents_map = array();
```

Comment:
Is this missing cache invalidation? I'm trying to think of a case where the lack of it would cause problems. If this `$dependents_map` is populated during `do_items()` at `wp_head`, and then additional scripts are enqueued in the body for printing at `wp_footer`, I suppose it doesn't matter that `$dependents_map` may be no longer current. For example:

```php
add_action( 'wp_enqueue_scripts', static function () {
    wp_enqueue_script( 'foo', '/foo.js', array(), null, array( 'strategy' => 'defer' ) );
} );

add_action( 'wp_body_open', static function () {
    wp_enqueue_script( 'bar', '/bar.js', array( 'foo' ), null, array( 'strategy' => 'defer' ) );
} );
```

By the time we get to `wp_footer`, the `$dependents_map` will not be reflecting that `bar` depends on `foo` since `foo` was already processed.

Since `$dependents_map` is private, we don't need to worry about other plugins relying on the mapping. So I don't think we _have_ to do anything here, but I just wanted to call out that it can become stale.

Reply:
I think you are right that there is a potential edge case where a late registered blocking script would not be available in the `$dependents_map` whenever we're evaluating the eligible strategies (which is the only place in the code where getting a list of dependents is needed). I'm not sure there is a way to totally solve that problem, since we can't anticipate the presence of a script before it's registered.

That said, we should probably look into populating this mapping during registration rather than doing so the first time the dependents for a handle is returned. I think that exploration can happen during beta.